pkg/credentialprovider/plugins.go

```diff
@@ -44,6 +55,10 @@ func RegisterCredentialProvider(name string, provider DockerConfigProvider) {
 	providers[name] = provider
 }
 
+func isDeprecatedInternal(name string) bool {
+	return deprecatedCredentialProviders.Has(name)
+}
+
 // NewDockerKeyring creates a DockerKeyring to use for resolving credentials,
 // which draws from the set of registered credential providers.
 func NewDockerKeyring() DockerKeyring {
```

Comment:
this is called every time every a pod is created with dockershim... this seems late/frequent to be warning about disabled credential providers

Reply:
I tried logging the warning in the providers' `init()` functions, but it turns out that's too early -- klog has not yet been initialized at that point, so the log level hasn't been set yet which means the the log never gets printed even with `--v=4`. I ended up using a `sync.Once` to avoid logging it repeatedly which feels a little clunky but works.